PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -1,49 +1,265 @@
-        // TODO: configure and start paypal native checkout
-        callback.onResult(new BraintreeException("TODO: integrate native checkout with Braintree SDK."));
+    private void startPayPalNativeCheckout(final FragmentActivity activity, final Configuration configuration, final String payPalClientId, final PayPalRequest payPalRequest, final PayPalNativeTokenizeCallback callback) {
+        internalPayPalClient.sendRequest(activity, payPalRequest, new PayPalInternalClientCallback() {
+            @Override
+            public void onResult(final PayPalResponse payPalResponse, final Exception error) {
+                if (payPalResponse != null) {
+                    startNativeCheckout(activity, configuration, payPalClientId, payPalResponse, callback);
+                } else {
+                    callback.onResult(null, error);
+                }
+            }
+        });
+    }
+
+    private void startNativeCheckout(final FragmentActivity activity, final Configuration configuration, final String clientId, final PayPalResponse payPalResponse, final PayPalNativeTokenizeCallback callback) {
+        final String redirectUrl = getPayPalReturnUrl(activity);
+        Environment environment = Environment.SANDBOX;
+        if ("production".equalsIgnoreCase(configuration.getEnvironment())) {
+            environment = Environment.LIVE;
+        }
+
+        CheckoutConfig checkoutConfig = new CheckoutConfig(
+                activity.getApplication(),
+                clientId,
+                environment,
+                redirectUrl
+        );
+        PayPalCheckout.setConfig(checkoutConfig);
+        PayPalCheckout.start(
+                new CreateOrder() {
+                    @Override
+                    public void create(final CreateOrderActions createOrderActions) {
+                        final String pairingId = payPalResponse.getPairingId();
+                        if (payPalResponse.isBillingAgreement()) {
+                            createOrderActions.setBillingAgreementId(pairingId);
+                        } else {
+                            createOrderActions.set(pairingId);
+                        }
+                    }
+                },
+                new OnApprove() {
+                    @Override
+                    public void onApprove(final Approval approval) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.approved");
+                        handleNativeCheckoutApproval(approval, payPalResponse, callback);
+                    }
+                },
+                new OnCancel() {
+                    @Override
+                    public void onCancel() {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.client_cancel");
+                        callback.onResult(null, new BraintreeException("Canceled"));
```

Comment:
Since this PR was last reviewed, we updated the exception type for cancellations to `UserCanceledException`, so that should be returned here now instead.